### DISCUSSION TITLE
how should I duplicate an element (shape)

### DISCUSSION THREAD

**Discussion by Delanoor** [pikasojs/pikaso#110] [Q&A] [answered]
Hello, 

I wanted to know if there is any way to duplicate a selected shape, and insert as a duplicated element, 

so for example, I have been trying something like
```
  const handleDuplicate = useCallback(() => {

    const selectedShapeData = selectedShape?.node.attrs // or clone()

    editor.board.addShape(selectedShape);
  }, [editor, selectedShape]);
```

but just could not find any way to achieve such
thank you very much

**Comment by raminious** [comment, marked as answer]
This feature is not implemented yet 
https://github.com/pikasojs/pikaso/issues/33

**Reply by Delanoor** [reply]
thank you, perhaps I'll contribute too

**Reply by raminious** [reply]
I do appreciate it
